Deploy Loki

alexandre-allard opened the pull request:
**Component**: salt, build, log

**Context**:

#2682 

**Summary**:

- Add support to remove manifest when rendering charts from `render.py` script
- Create a new storage class for Loki `metalk8s-loki`
- Take Loki charts from https://grafana.github.io/loki/charts and adapt value file to metalk8s
- Render the Loki chart to salt states
- Add CSC handling for Loki configuration
- Add simple test to check that Loki work
- Add new example file for Loki volume and use it in CI and vagrant deployment

**Acceptance criteria**: 


---

Closes: #2682
Refs: #2712

alexandre-allard committed:
charts/log: initial import of loki helm charts

  helm repo add loki https://grafana.github.io/loki/charts
  helm repo update
  helm fetch -d charts --untar loki/loki

Refs: #2682

alexandre-allard committed:
charts/log: add loki helm configuration values

this file is used to overwrite default values.

Refs: #2682

alexandre-allard committed:
build: fetch loki image during build

Refs: #2682

alexandre-allard committed:
salt: render loki chart into Salt formula

  ./charts/render.py loki --namespace metalk8s-logging \
    --remove-manifest Secret loki \
    charts/loki.yaml charts/loki/ \
    > salt/metalk8s/addons/logging/loki/deployed/chart.sls

Refs: #2682

alexandre-allard committed:
build: add loki formulas to salt tree

Refs: #2682

alexandre-allard committed:
salt: deploy metalk8s-logging namespace

Refs: #2682

alexandre-allard committed:
salt: deploy logging stack during MetalK8s install

For now it only deploys loki, but it'll
also include Fluent Bit installation.

Refs: #2682

alexandre-allard committed:
charts: add option to remove manifests

Add `--remove-manifest KIND NAME` option allowing
to remove some manifests from the resulting chart.
Useful when we want to deploy and manage ourself
some object of an addon (e.g. Loki secret).

Refs: #2682

alexandre-allard committed:
salt: add CSC mechanism to Loki

Re-render the chart with CSC:

  ./charts/render.py loki \
    --remove-manifest Secret loki \
    --namespace metalk8s-logging \
    --service-config loki metalk8s-loki-config \
    metalk8s/addons/logging/loki/config/loki.yaml \
    charts/loki.yaml charts/loki/ \
    > salt/metalk8s/addons/logging/loki/deployed/chart.sls

Refs: #2682

alexandre-allard committed:
salt: add Loki storageclass

Refs: #2682

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

TeddyAndrieux committed:
tests: Add tests for Loki alone

Add test about Loki only, to check that Loki work properly, this test
basically just use Loki API
(https://grafana.com/docs/loki/latest/api/#get-lokiapiv1query) to push a
Simple test log and then query this pushed log from Loki

Refs: #2712

TeddyAndrieux committed:
charts: add persistence storage to Loki

Enable persistence from Loki chart
NOTE: That Loki chart does not allow to provide selector to claim
Volume, so any volume using `metalk8s-loki` storageClass can be bound to
the Loki container

Refs: #2682

TeddyAndrieux committed:
build,example,eve: Add example YAML for Loki volume

Add an example file to create a sparseLoopDevice volume for Loki, add
this file in the ISO and also use it for CI and vagrant deployment

Refs: #2682

alexandre-allard committed:
charts: add option to change path of CSC defaults

This is needed as we put Loki inside logging directory,
so the path is not metalk8s/addons/loki/config/loki.yaml
but metalk8s/addons/logging/loki/config/loki.yaml, so
we need to be able to set the path.

Refs: #2682

TeddyAndrieux reviewed (approved):
LGTM

alexandre-allard commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
